dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp

```diff
@@ -1637,19 +1640,65 @@ TEST(DataInterfaceIIDMTest, testImportError) {
 }
 
 TEST(DataInterfaceIIDMTest, testImportExport) {
-  const powsybl::iidm::Network& network = createNodeBreakerNetworkIIDM();
+  auto network = createNodeBreakerNetworkIIDM();
+
+  shared_ptr<DataInterfaceIIDM> dataOutput = createDataItfFromNetwork(createNodeBreakerNetworkIIDM());
+  ASSERT_NO_THROW(dataOutput->dumpToFile("network.xml"));
+  const powsybl::iidm::Network& outputNetwork = dataOutput->getNetworkIIDM();
+  ASSERT_THROW_DYNAWO(dataOutput->dumpToFile(".."), Error::GENERAL, KeyError_t::XmlFileParsingError);
+
+  shared_ptr<DataInterface> dataInput = DataInterfaceIIDM::build("network.xml");
+  shared_ptr<DataInterfaceIIDM> dataInputIIDM = boost::dynamic_pointer_cast<DataInterfaceIIDM>(dataInput);
+  const powsybl::iidm::Network& inputNetwork = dataInputIIDM->getNetworkIIDM();
+
+  ASSERT_EQ(outputNetwork.getId(), inputNetwork.getId());
+  ASSERT_EQ(outputNetwork.getId(), network->getId());
+  ASSERT_EQ(inputNetwork.getId(), network->getId());
+}
+
+TEST(DataInterfaceIIDMTest, testMultiThreading) {
+  auto network = createNodeBreakerNetworkIIDM();
 
   shared_ptr<DataInterfaceIIDM> dataOutput = createDataItfFromNetwork(createNodeBreakerNetworkIIDM());
   ASSERT_NO_THROW(dataOutput->dumpToFile("network.xml"));
   const powsybl::iidm::Network& outputNetwork = dataOutput->getNetworkIIDM();
   ASSERT_THROW_DYNAWO(dataOutput->dumpToFile(".."), Error::GENERAL, KeyError_t::XmlFileParsingError);
 
   shared_ptr<DataInterface> dataInput = DataInterfaceIIDM::build("network.xml");
+  ASSERT_FALSE(dataInput->canUseVariant());
+  dataInput = DataInterfaceIIDM::build("network.xml", 2);
+  ASSERT_TRUE(dataInput->canUseVariant());
   shared_ptr<DataInterfaceIIDM> dataInputIIDM = boost::dynamic_pointer_cast<DataInterfaceIIDM>(dataInput);
   const powsybl::iidm::Network& inputNetwork = dataInputIIDM->getNetworkIIDM();
 
   ASSERT_EQ(outputNetwork.getId(), inputNetwork.getId());
-  ASSERT_EQ(outputNetwork.getId(), network.getId());
-  ASSERT_EQ(inputNetwork.getId(), network.getId());
+  ASSERT_EQ(outputNetwork.getId(), network->getId());
+  ASSERT_EQ(inputNetwork.getId(), network->getId());
+
+  auto load_interface_ptr = dataInput->getNetwork()->getVoltageLevels()[0]->getLoads().front();
+  auto load_iterface_iidm_ptr = boost::dynamic_pointer_cast<LoadInterfaceIIDM>(load_interface_ptr);
+  auto& load = load_iterface_iidm_ptr->getUnderlyingLoad();
+
+  std::thread launch0([&dataInput, &load](){
+    dataInput->useVariant("0");
+    load.setP0(3.0);
```

Comment:
This is setting directly the IIDM so it's not really a test of the usage of data itf in MT mode

Reply:
No it is a unit test to check the multi trheading variant system from ata interface POV

Reply:
ok, maybe we could add more tests here then?

Reply:
I'll add a test to check clone